core/src/worker/slot_provider.rs

```diff
@@ -130,12 +130,9 @@ mod tests {
         if let Some(slot) = provider.try_reserve_wft_slot() {
             let p = slot.schedule_wft(new_validatable_response());
             assert!(p.is_ok());
-
-            // This is a compiler error, schedule_wft consumes the slot...
-            // let p = slot.schedule_wft(new_validatable_response());
-            // assert!(p.is_err());
-
             assert!(external_wft_rx.recv().await.is_some());
+        } else {
+            panic!("failed to reserve slot");
```

Comment:
You can just use `.expect()` on the option now

Reply:
Much clear, thanks!